src/main/java/io/nats/client/Options.java

```diff
@@ -402,6 +403,11 @@ public class Options {
      * {@link Builder#serverPool(ServerPool) serverPool}.
      */
     public static final String PROP_SERVERS_POOL_IMPLEMENTATION_CLASS_PREFERRED = "servers.pool.implementation.class";
+    /**
+     * Property used to set class name for the Dispatcher Factory
+     * {@link Builder#dispatcherFactory(DispatcherFactory) dispatcherFactory}.
+     */
+    public static final String PROP_DISPATCHER_FACTORY_CLASS = "dispatcher.factory.class";

```

Comment:
for options, there is a build, but you can also set factory class names via property